File: FinalProjectCalendar/app/src/main/java/com/example/finalprojectcalendar/MainActivity.java
package com.example.finalprojectcalendar;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Build;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import java.time.LocalDate;
import java.time.YearMonth;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    //Navigation
    private BottomNavigationView bottomNav;

    //Monthly Calendar


    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        if(AppCompatDelegate.getDefaultNightMode() == AppCompatDelegate.MODE_NIGHT_YES){
            setTheme(R.style.DarkTheme);
        }
        else{
            setTheme(R.style.AppTheme);
        }
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);



        bottomNav = findViewById(R.id.bottomNav);
        bottomNav.setOnNavigationItemSelectedListener(navListener);
        getSupportFragmentManager().beginTransaction().replace(R.id.container, new MonthView()).commit();

        //if savedinstance


    }




    private BottomNavigationView.OnNavigationItemSelectedListener navListener =
            new BottomNavigationView.OnNavigationItemSelectedListener() {
                @Override
                public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                    Fragment selectedFragment = null;
                    //View view = bottomNav.findViewById(item.getItemId());
                    //view.performClick();

                    switch (item.getItemId()) {
                        case R.id.nav_month:
                            selectedFragment = new MonthView();
                            break;

                        case R.id.nav_week:
                            selectedFragment = new WeeklyView();
                            break;

                        case R.id.nav_events:
                            selectedFragment = new EventView();
                            break;

                        case R.id.nav_settings:
                            selectedFragment = new SettingsView();
                            break;
                    }

                    getSupportFragmentManager().beginTransaction().replace(R.id.container, selectedFragment).commit();

                    return false;
                }
            };




}

File: FinalProject/app/src/main/java/com/example/finalproject/MainActivity.kt
package com.example.finalproject

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.example.finalproject.Fragments.EventView
import com.example.finalproject.Fragments.MonthView
import com.example.finalproject.Fragments.WeeklyView
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {

    private val monthlyFragment = MonthView()
    private val weeklyFragment = WeeklyView()
    private val eventFragment = EventView()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        replaceFragment(monthlyFragment)

        val bottomNavigation = findViewById<BottomNavigationView>(R.id.bottomNav)
        bottomNavigation.setOnNavigationItemSelectedListener(bottomNavMethod)



    }

    private fun replaceFragment(fragment: Fragment) {
        if(fragment != null){
            val transaction = supportFragmentManager.beginTransaction()
            transaction.replace(R.id.mainFrame, fragment)
            transaction.commit()
        }
    }

    private val bottomNavMethod =
        BottomNavigationView.OnNavigationItemSelectedListener { menuItem ->
            var bottomFragment: Fragment = monthlyFragment
            when (menuItem.itemId) {
                R.id.month -> bottomFragment = monthlyFragment
                R.id.week -> bottomFragment = weeklyFragment
                R.id.events -> bottomFragment = eventFragment
            }
            replaceFragment(bottomFragment)
            true
        }

}